packages/enhanced/src/lib/container/runtime/FederationModulesPlugin.ts

```diff
@@ -0,0 +1,55 @@
+import type { Compiler, Compilation as CompilationType } from 'webpack';
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
+const Compilation = require(
+  normalizeWebpackPath('webpack/lib/Compilation'),
+) as typeof import('webpack/lib/Compilation');
+import { SyncHook } from 'tapable';
+
+/** @type {WeakMap<Compilation, CompilationHooks>} */
+const compilationHooksMap = new WeakMap<CompilationType, CompilationHooks>();
+
+const PLUGIN_NAME = 'FederationModulesPlugin';
+
+/** @typedef {{ header: string[], beforeStartup: string[], startup: string[], afterStartup: string[], allowInlineStartup: boolean }} Bootstrap */
+
+type CompilationHooks = {
+  getContainerEntryModules: SyncHook<[any], void>;
+};
+
+class FederationModulesPlugin {
+  /**
+   * @param {Compilation} compilation the compilation
+   * @returns {CompilationHooks} the attached hooks
+   */
+  static getCompilationHooks(compilation: CompilationType): CompilationHooks {
```

Comment:
based off JavascriptModulesPlugin design so it can be tapped for multiple compilations at once. 
This allows sharing of module info that is painful to recreate or find again.